### ISSUE TITLE
some question about function "calc_topo_weights_with_components_idx"

### ISSUE THREAD

**Issue by jingzhengli** [pxiangwu/TopoFilter#3]
Hi, thanks for the nice code.
In the **train.py**, some lines confuse me as follows:

lines 277,283 and 284, define some vectors

         prob_all[indices] = softmax_outputs.detach().cpu()
         train_gt_labels[indices] = labels.cpu().numpy().astype(np.int64)
         train_pred_labels[indices] = labels.cpu().numpy().astype(np.int64)

why the "**train_gt_labels**" is the same as "**train_pred_labels**" both of which are taken as input to the function **calc_topo_weights_with_components_idx**? and the "**prob_all**" is defined but not used.

**Comment by pxiangwu** [comment]
Hi, for your question:

> why the "train_gt_labels" is the same as "train_pred_labels" both of which are taken as input to the function

This is due to the function signature of `calc_topo_weights_with_components_idx`. In particular, the code of `calc_topo_weights_with_components_idx` has not been cleansed and is a bit messy. So currently I simply feed `train_gt_labels` and `train_pred_labels` together to comply with the function signature of `calc_topo_weights_with_components_idx`.

> the "prob_all" is defined but not used

Thank you for pointing out this. Yes, it is not used. I used this variable for debugging purpose, but forgot to delete it in the code ...

**Comment by jingzhengli** [comment]
Thanks for your quick reply. I got it, I just make sure it's the right.
